cou/steps/execute.py

```diff
@@ -59,7 +60,20 @@ async def _run_step(step: BaseStep, prompt: bool, overwrite_progress: bool = Fal
         logger.debug("running all sub-steps of %s step sequentially", step)
         for sub_step in step.sub_steps:
             logger.debug("running sub-step %s of %s step", sub_step, step)
-            await apply_step(sub_step, prompt, overwrite_substeps_progress)
+            try:
+                await apply_step(sub_step, prompt, overwrite_substeps_progress)
+            except HaltUpgradeExecution:
+                sub_step_index = step.sub_steps.index(sub_step)
+                non_executed_sub_steps = [
+                    sub_step.description for sub_step in step.sub_steps[sub_step_index + 1 :]
+                ]
+                logger.warning(
+                    "Step: '%s' failed to complete execution. Steps '%s' will be skipped.",
+                    sub_step.description,
+                    ", ".join(non_executed_sub_steps),
```

Comment:
not-blocking: I think that this will not be readable for user, and we should use new line, but this can be improved during manual test phase.